src/models/userModel.js

```diff
@@ -0,0 +1,47 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+
+const Schema = mongoose.Schema;
+
+let userSchema = new Schema({
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  firstName: {
+    type: String,
+    required: true,
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  role: {
+    type: String,
+    default: "ROLE_USER",
+  },
+});
+
+// Pre save to hash password
+userSchema.pre("save", function preSave(next) {
+  const user = this;
+
+  if (!user.isModified("password")) return next();
+
+  bcrypt
+    .hash(user.password, 10)
+    .then((hash) => {
+      user.password = hash;
+      next();
+    })
+    .catch((error) => {
+      return next(error);
```

Comment:
I'm afraid that if you do a next, the data is still added to the database but not hashed? I'm not sure of what I'm saying, check it out.

Reply:
I'll test it tonight